plugins/UsersManager/API.php

```diff
@@ -107,7 +107,16 @@ public function getUserPreference($userLogin, $preferenceName)
             return $optionValue;
         }
 
-        return $this->getDefaultUserPreference($preferenceName, $userLogin);
+        $defaultPreference = $this->getDefaultUserPreference($preferenceName, $userLogin);
+
+        // A `get*` method should not set anything. But in this case we do it to improve performance as it prevents
+        // us from loading all siteIds (which can be 50k or more) the next time this preference is requested.
+        // this method can be called quite often when generating links etc (to get defaultWebsiteId).
+        if ($defaultPreference !== false && $preferenceName === self::PREFERENCE_DEFAULT_REPORT) {
+            $this->setUserPreference($userLogin, $preferenceName, $defaultPreference);
+        }
```

Comment:
> Would that be an issue for the time before the scheduled task is run?

It would be only a bit poor-er performance. Should be ok and still better than currently. 

> I will also default the user preference when creating/updating users.

OK in case it is easy to do otherwise not needed I reckon. It is easy to implement but when also writing tests etc then it always takes time... When creating a user wouldn't make much sense if I get it right as a user wouldn't have permission to any site yet.